source/content.js

```diff
@@ -1,317 +1,100 @@
-import focusConfirmationButtons from './features/focus-confirmation-buttons';
-import addKeyboardShortcutsToCommentFields from './features/add-keyboard-shortcuts-to-comment-fields';
-import addCreateReleaseShortcut from './features/add-create-release-shortcut';
-import addCILink from './features/add-ci-link';
-import embedGistInline from './features/embed-gist-inline';
-import extendStatusLabels from './features/extend-status-labels';
-import highlightClosingPrsInOpenIssues from './features/highlight-closing-prs-in-open-issues';
-import addJumpToBottomLink from './features/add-jump-to-bottom-link';
-import addQuickReviewButtons from './features/add-quick-review-buttons';
-import extendDiffExpander from './features/extend-diff-expander';
-import sortIssuesByUpdateTime from './features/sort-issues-by-update-time';
-import makeDiscussionSidebarSticky from './features/make-discussion-sidebar-sticky';
-import makeHeadersSticky from './features/make-headers-sticky';
-import shortenLinks from './features/shorten-links';
-import waitForBuild from './features/wait-for-build';
-import addDownloadFolderButton from './features/add-download-folder-button';
-import hideUselessNewsfeedEvents from './features/hide-useless-newsfeed-events';
-import closeOutOfViewModals from './features/close-out-of-view-modals';
-import monospaceTextareas from './features/monospace-textareas';
-import improveShortcutHelp from './features/improve-shortcut-help';
-import hideNavigationHoverHighlight from './features/hide-navigation-hover-highlight';
-import addPullRequestHotkey from './features/add-pull-request-hotkey';
-import openSelectionInNewTab from './features/add-selection-in-new-tab';
-import addSwapBranchesOnCompare from './features/add-swap-branches-on-compare';
-import showFollowersYouKnow from './features/show-followers-you-know';
-import hideCommentsFaster from './features/hide-comments-faster';
-import linkifyCommitSha from './features/linkify-commit-sha';
-import hideIssueListAutocomplete from './features/hide-issue-list-autocomplete';
-import showUserTopRepositories from './features/show-user-top-repositories';
-import userProfileFollowerBadge from './features/user-profile-follower-badge';
-import usefulNotFoundPage from './features/useful-not-found-page';
-import setDefaultRepositoriesTypeToSources from './features/set-default-repositories-type-to-sources';
-import markPrivateOrgs from './features/mark-private-orgs';
-import navigatePagesWithArrowKeys from './features/navigate-pages-with-arrow-keys';
-import addStarRepoHotkey from './features/add-star-repo-hotkey';
-import bypassChecksTravis from './features/bypass-checks-travis';
-
-import * as pageDetect from './libs/page-detect';
-import {safeElementReady, enableFeature, safeOnAjaxedPages, injectCustomCSS} from './libs/utils';
-
-// Add globals for easier debugging
+import 'webext-dynamic-content-scripts';
+import features from './libs/features';
+
+import './features/useful-not-found-page';
```

Comment:
Can we sort all of these lines, please!!!

![](https://media.giphy.com/media/aZ4sQUpybai5y/giphy.gif)

Reply:
Can't, some lines must come after others. The order was preserved from how it was before 😅

Reply:
We could separate the dependencies and the dependent features into two sections separated by a line.

Reply:
I'm (a bit) against this.

1. There are features that need to be imported after another.
2. When adding a new feature you need to sort it in again instead of just adding it to the end.

Reply:
> There are features that need to be imported after another.

Yeah, that's why I proposed that we could separate all the imports intontwo sections

``` js
// imports that will be depended upon
import '...';

// usual imports
import '...';
```

That way, we could still maintain the sorted order.

> When adding a new feature you need to sort it in again instead of just adding it to the end.

That should not be a big of an issue, because most editors let you sort lines easily

Reply:
As far as I know, only 2 features have "dependents" and they can just be pushed to the end of the file, so I suppose that's the easiest solution for now.

**However** let's skip the order **for this PR.** Most features have awkward `add-*` names that when ordered wouldn't make much of a difference anyway.

In a separate commit I can just rename the files and the diff will be empty (just a lot of `rename`)